From fa137e3b5c994508370e0cd2396ece081a1316c4 Mon Sep 17 00:00:00 2001
From: Christopher Faulet <cfaulet@haproxy.com>
Date: Wed, 27 Nov 2019 14:00:51 +0100
Subject: [PATCH 1/1] BUG/MINOR: h1: Don't test the host header during
 response parsing

During the H1 message parsing, the host header is tested to be sure it matches
the request's authority, if defined. When there are multiple host headers, we
also take care they are all the same. Of course, these tests must only be
performed on the requests. A host header in a response has no special meaning.

This patch must be backported to 2.1.

(cherry picked from commit bc7c03eba39ec9f0b94734399853bbece1e1a250)
Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>
---
 src/h1.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/h1.c b/src/h1.c
index 83afb14..15827db 100644
--- a/src/h1.c
+++ b/src/h1.c
@@ -833,7 +833,7 @@ int h1_headers_to_hdr_list(char *start, const char *stop,
 						break;
 					}
 				}
-				else if (isteqi(n, ist("host"))) {
+				else if (!(h1m->flags & H1_MF_RESP) && isteqi(n, ist("host"))) {
 					if (host_idx == -1) {
 						struct ist authority;
 
-- 
1.7.10.4

